bigquery-storage/api/BigQueryStorage.Samples/AppendRowsPending.cs

```diff
@@ -0,0 +1,141 @@
+        ProtoData protoData = new ProtoData
+        {
+            // This is a TEMPORARY WORKAROUND via extension method using reflection till the feature request
+            // and PR for "ToProto()" in Protobuf library gets released.
+            WriterSchema = new ProtoSchema { ProtoDescriptor = CustomerRecord.Descriptor.ToProto() },
+            Rows = new ProtoRows
+            {
+                SerializedRows =
+                {
+                    new CustomerRecord{ CustomerNumber = 1, CustomerName = "Alice"}.ToByteString(),
+                    new CustomerRecord{ CustomerNumber = 2, CustomerName = "Bob"}.ToByteString()
+                }
+            }
+        };
+
+        // Initialize the append row request
+        // Set an offset to allow resuming this stream if the connection breaks.
+        // Keep track of which requests the server has acknowledged and resume the
+        // stream at the first non-acknowledged message. If the server has already
+        // processed a message with that offset, it will return an ALREADY_EXISTS
+        // error, which can be safely ignored.
+        // The first request must always have an offset of 0.
+        AppendRowsRequest appendRowrequest = new AppendRowsRequest
+        {
+            WriteStreamAsWriteStreamName = stream.WriteStreamName,
+            Offset = 0L,
+            ProtoRows = protoData
+        };
+
+        // Stream a request to the server
+        await rowAppender.WriteAsync(appendRowrequest);
+
+        protoData.Rows = new ProtoRows
+        {
+            SerializedRows =
+            {
+                new CustomerRecord { CustomerNumber = 3, CustomerName = "Charles"}.ToByteString()
+            }
+        };
+
+        // Since this is the second request, you only need to include the row data.
+        // The name of the stream and protocol buffers DESCRIPTOR is only needed in
+        // the first request.
+        appendRowrequest = new AppendRowsRequest
+        {
+            Offset = 2L, //Offset must equal the number of rows that were previously sent.
+            ProtoRows = protoData
+        };
+
+        await rowAppender.WriteAsync(appendRowrequest);
+
+        // Complete writing requests to the stream
+        await rowAppender.WriteCompleteAsync();
+
+        // Await the handler. This will complete once all server responses have been processed
+        await appendResultsHandlerTask;
+
+        // A PENDING type stream must be "finalized" before being committed. No new
+        // records can be written to the stream after this method has been called.
+        await bigQueryWriteClient.FinalizeWriteStreamAsync(stream.Name);
+        BatchCommitWriteStreamsRequest batchCommitWriteStreamsRequest = new BatchCommitWriteStreamsRequest
+        {
+            Parent = table.ToString(),
+            WriteStreams = { stream.Name }
+        };
+
+        await bigQueryWriteClient.BatchCommitWriteStreamsAsync(batchCommitWriteStreamsRequest);
```

Comment:
Shouldn't we be checking for errors in the response?

Reply:
Tried incorporating checking error with my limited understanding. Please review and suggest to make it fit for sample code.